### PR TITLE
Delete the concept of "location" values from local morph

### PR DESCRIPTION
**Pull request by SingleAccretion**
No longer needed after ADDR's removal.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=258386&view=ms.vss-build-web.run-extensions-tab) - none in code, `0.1%`-`0.15%` TP wins (up to `0.25%` for x86+MinOpts).

Linux x64 failure looks like #85371.

@dotnet/jit-contrib

File: src/coreclr/jit/lclmorph.cpp [hunk 1]
```diff
@@ -382,41 +327,26 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
...
-        //    to escape the input value.
-        //
-        // Notes:
-        //   - LOCATION(lclNum, offset) => not representable, must escape
-        //   - ADDRESS(lclNum, offset) => LOCATION(lclNum, offset + addOffset)
-        //     if the offset overflows then location is not representable, must escape
-        //   - UNKNOWN => UNKNOWN
-        //
-        bool Indir(Value& val, unsigned addOffset = 0)
+        bool AddOffset(unsigned addOffset)
```
Comment (SingleAccretion) [inline src/coreclr/jit/lclmorph.cpp:338]: A bit of code duplication (and no header comment), but this will be deleted in #85568.

**Review by AndyAyersMS — APPROVED** [review]
LGTM, just a commenting nit.

**Comment by AndyAyersMS** [comment]
Thanks again for your work to make the jit better.

**Pull request merged**